presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusClient.java

```diff
@@ -148,17 +169,61 @@ public byte[] fetchUri(URI uri)
     {
         Request.Builder requestBuilder = new Request.Builder().url(uri.toString());
         getBearerAuthInfoFromFile().map(bearerToken -> requestBuilder.header("Authorization", "Bearer " + bearerToken));
-
         Response response;
         try {
-            response = httpClient.newCall(requestBuilder.build()).execute();
-            if (response.isSuccessful() && response.body() != null) {
-                return response.body().bytes();
+            if (config.isTlsEnabled()) {
+                OkHttpClient httpClient;
+                HostnameVerifier hostnameVerifier = new HostnameVerifier() {
+                    @Override
+                    public boolean verify(String hostname, SSLSession session)
+                    {
+                        // Always return true to accept any hostname
+                        return true;
```

Comment:
Isn't there any alternative to trusting all hosts?

Reply:
This is only using based on VerifyHostName external value, otherwise trusting all hosts
if (!config.getVerifyHostName()) {
                    httpClient = new OkHttpClient.Builder()
                            .hostnameVerifier(hostnameVerifier)
                            .sslSocketFactory(getSSLContext().getSocketFactory(), (X509TrustManager) getTrustManagerFactory().getTrustManagers()[0])
                            .build();
                }